### PR TITLE
Added data layer to VAMC featured content links

### PR DESCRIPTION
**Pull request by gescobardev**
## Description
Linked issue below

To preview- http://localhost:3002/preview?nodeId=1507

Example of what is expected after you click one of the links in screenshot below:
<details>
<summary>Screenshot</summary>

Screen Shot 2021-06-28 at 3 28 42 PM

</details>

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by brangi — APPROVED** [review]

**Review by k80bowman — APPROVED** [review]

File: src/site/paragraphs/list_of_link_teasers_facility.drupal.liquid [hunk 1]
```diff
@@ -33,7 +33,9 @@
             {% if forloop.first || everyStartItem == 0 %}
                 <div class="usa-width-one-half vads-u-display--flex vads-u-flex-direction--column">
             {% endif %}
-                    <a onClick="recordEvent({ event: 'nav-featured-content-link-click', 'featured-content-header': 'In the Spotlight' });" class="vads-u-margin-bottom--2" href="{{link.url.path}}">
+                    <a onClick="recordEvent({ event: 'nav-featured-content-link-click', 'featured-content-header': 'In the Spotlight', 'featured-content-click-label': '{{ link.title }}' });"
```
Comment (mpelzsherman) [inline src/site/paragraphs/list_of_link_teasers_facility.drupal.liquid:36]: The AC say:

> featured-content-heading': //dynamically populate according to the header of the featured content

Should we pass `paragraph.fieldTitle` here?

Comment (gescobardev) [reply src/site/paragraphs/list_of_link_teasers_facility.drupal.liquid:36]: @mpelzsherman  Sure I can pass that in. Unless theres a need for it to specifically say `In the spotlight` ?
Passing it in would make it specific for example- `In the spotlight at VA Pittsburgh health care`

**Comment by gescobardev** [comment]
I will also add `list_of_link_teasers_facility.drupal.liquid` to facilities in `CODEOWNERS`

File: src/site/paragraphs/list_of_link_teasers_facility.drupal.liquid (hunk 1 above)
Comment (mpelzsherman) [reply src/site/paragraphs/list_of_link_teasers_facility.drupal.liquid]: That was how I interpreted the AC. You may want to confirm with @mmiddaugh or someone from the analytics team.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -49,6 +49,7 @@ src/site/includes/vet_centers @department-of-veterans-affairs/vsa-facilities-fro
 src/site/components/phone.drupal.liquid @department-of-veterans-affairs/vsa-facilities-frontend
 src/site/components/phone-number.drupal.liquid @department-of-veterans-affairs/vsa-facilities-frontend
 src/site/teasers @department-of-veterans-affairs/vsa-facilities-frontend
+src/site/paragraphs/list_of_link_teasers_facility.drupal.liquid @department-of-veterans-affairs/vsa-facilities-frontend
```
Comment (mpelzsherman) [inline .github/CODEOWNERS:52]: Let's create a "facilities" subfolder under `src/site/paragraphs`, and move `list_of_link_teasers_facility.drupal.liquid` into it. Then we can add `src/site/paragraphs/facilities` here instead, so any new facilities-specific files have a place to live.

Comment (gescobardev) [reply .github/CODEOWNERS]: sounds good to me!

**Review by mpelzsherman — APPROVED** [review]

**Pull request merged**